test/extended/operators/check_crd_status.go

```diff
@@ -0,0 +1,113 @@
+package operators
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	"k8s.io/kube-openapi/pkg/util/sets"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// checkCRDs returns a list of names of CRDs that don't have subresource.status
+// For now, it ignores the ones that are currently failing.
+func checkCRDs(crdItemList []v1.CustomResourceDefinition) []string {
+
+	// These 17 CRDs that, at the time this test was written, do not have subresource.status.
+	// These can be skipped for now but we don't want the number to increase.
+	//
+	exceptionsList := sets.NewString(
+		"clusternetworks.network.openshift.io",
+		"consoleplugins.console.openshift.io",
+		"consolequickstarts.console.openshift.io",
+		"consoleyamlsamples.console.openshift.io",
+		"egressnetworkpolicies.network.openshift.io",
+		"hostsubnets.network.openshift.io",
+		"machineconfigs.machineconfiguration.openshift.io",
+		"netnamespaces.network.openshift.io",
+		"networks.config.openshift.io",
+		"networks.operator.openshift.io",
+		"nodes.config.openshift.io",
+		"operatorpkis.network.operator.openshift.io",
+		"profiles.tuned.openshift.io",
+		"rangeallocations.security.internal.openshift.io",
+		"rolebindingrestrictions.authorization.openshift.io",
+		"securitycontextconstraints.security.openshift.io",
+		"tuneds.tuned.openshift.io",
+	)
+
+	failures := []string{}
+	for _, crdItem := range crdItemList {
+
+		// This test is interested only in CRDs that end with "openshift.io".
+		if !strings.HasSuffix(crdItem.ObjectMeta.Name, "openshift.io") {
+			continue
+		}
+
+		crdName := crdItem.ObjectMeta.Name
+
+		// Skip CRDs in the exceptions list for now.
+		if exceptionsList.Has(crdName) {
+			continue
+		}
+
+		// Iterate through all versions of the Spec.
+		var versionName string
+		for i := 0; i < len(crdItem.Spec.Versions); i++ {
```

Comment:
you only need to do this check if the openapi spec has a status element.

Reply:
If `CustomResourceDefinition.Status` was a pointer, I can do `if crdItem.Status ~= nil`.

But since it's not a pointer, that syntax is illegal so I don't know how to tell if it exists or not.

Reply:
https://stackoverflow.com/questions/28447297/how-to-check-for-an-empty-struct should do the trick, essentially comparing the Status struct to an empty zero value'd Status struct, if equal, it's unset.

Reply:
thanks @dgoodwin a straight composite literal compare wouldn't compile because of the slices and struct (with its own slice) inside `CustomResourceDefinitionStatus` so I went into the `Status` struct to check on fields that would not be empty if the `status` element was set.